src/lib.rs

```diff
@@ -282,32 +284,39 @@ impl AppDetails {
             proxied: Some(true),
             proxied_by: Some(proxied_by.to_string()),
             acars_router_version: Some(acars_router_version.to_string()),
+            acars_router_uuid: Some(Uuid::new_v4().to_string()),
         }
     }
     /// Updates an existing entry of `AppDetails` with the provided details.
     /// ```
     /// use acars_vdlm2_parser::AppDetails;
-    /// let manual_vdlm2: AppDetails = AppDetails { name: "dumpvdl2".to_string(), ver: "2.2.0".to_string(), proxied: Some(true), proxied_by: Some("acars_router".to_string()), acars_router_version: Some("1.0.12".to_string()) };
-    /// let mut vdlm2: AppDetails = AppDetails { name: "dumpvdl2".to_string(), ver: "2.2.0".to_string(), proxied: None, proxied_by: None, acars_router_version: None };
-    /// let manual_acars: AppDetails = AppDetails { name: "acarsdec". to_string(), ver: "3.7".to_string(), proxied: Some(true), proxied_by: Some("acars_router".to_string()), acars_router_version: Some("1.0.12".to_string()) };
-    /// let mut acars: AppDetails = AppDetails { name: "acarsdec". to_string(), ver: "3.7".to_string(), proxied: None, proxied_by: None, acars_router_version: None };
+    /// let manual_vdlm2: AppDetails = AppDetails { name: "dumpvdl2".to_string(), ver: "2.2.0".to_string(), proxied: Some(true), proxied_by: Some("acars_router".to_string()), acars_router_version: Some("1.0.12".to_string()), acars_router_uuid: Some("00000000-0000-0000-0000-000000000000".to_string()) };
+    /// let mut vdlm2: AppDetails = AppDetails { name: "dumpvdl2".to_string(), ver: "2.2.0".to_string(), proxied: None, proxied_by: None, acars_router_version: None, acars_router_uuid: Some("00000000-0000-0000-0000-000000000000".to_string())  };
+    /// let manual_acars: AppDetails = AppDetails { name: "acarsdec". to_string(), ver: "3.7".to_string(), proxied: Some(true), proxied_by: Some("acars_router".to_string()), acars_router_version: Some("1.0.12".to_string()), acars_router_uuid: Some("00000000-0000-0000-0000-000000000000".to_string()) };
+    /// let mut acars: AppDetails = AppDetails { name: "acarsdec". to_string(), ver: "3.7".to_string(), proxied: None, proxied_by: None, acars_router_version: None, acars_router_uuid: Some("00000000-0000-0000-0000-000000000000".to_string())  };
     /// vdlm2.proxy("acars_router", "1.0.12");
     /// acars.proxy("acars_router", "1.0.12");
     /// assert_eq!(vdlm2, manual_vdlm2);
     /// assert_eq!(acars, manual_acars);
     /// ```
     pub fn proxy(&mut self, proxied_by: &str, acars_router_version: &str) {
+        let acars_router_uuid = match self.acars_router_uuid.is_some() {
```

Comment:
No need to do this, as you can get away with:

```
if self.acars_router_uuid.is_none() {
    self.acars_router_uuid = Some(Uuid::new_v4().to_string());
}
```

Gets rid of the need to do an `.unwrap()` and an extra assignment, and only sets it if it's not set already.